Verniana-Jules Verne Studies.js

```diff
@@ -31,51 +30,74 @@
 	<http://www.gnu.org/licenses/>.
 */
 
+// eslint-disable-next-line
+function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
 
-function detectWeb(doc, url) { return FW.detectWeb(doc, url); }
-function doWeb(doc, url) { return FW.doWeb(doc, url); }
+function detectWeb(doc, _url) {
+	if (doc.querySelector('ul.volume_toc li')) return 'multiple';
+	if (doc.querySelector('.ref_on_top')) return 'journalArticle';
+	return false;
+}
+
+function match(t, r) {
+	let m = t.match(r);
+	return m ? m[1] : '';
+}
+
+function scrape(doc, url) {
+	let item = new Zotero.Item('journalArticle');
+	item.url = url;
+	item.title = ZU.capitalizeTitle(ZU.trimInternal(text(doc, '#content h1') || '')) || 'Verniana — Jules Verne Studies';
+	item.creators = [ZU.cleanAuthor(text(doc, 'div#content p.author a') || text(doc, 'div#content p.author'), 'author')];
+	item.abstractNote = unescape(text(doc, 'p.abstract') || '');
 
-// Articles
-FW.Scraper ({
-	itemType : 'journalArticle',
-	detect : FW.Xpath('//div[@class="ref_on_top"]'),
-	title : FW.Xpath('//div[@id="content"]/h1[1]')
-						.text().trimInternal().capitalizeTitle(),
-	creators : FW.Xpath('//div[@id="content"]/p[@class="author"][1]/a | //div[@id="content"]/p[@class="author" and not(a)][1]')
-						.text().cleanAuthor('author'),
-	title : 'Verniana — Jules Verne Studies',
-	abstractNote : FW.Xpath('//p[@class="abstract"]').text().unescapeHTML(),
-	volume: FW.Xpath('//div[@class="ref_on_top"]')
-					.text().match(/volume (\d+)/i, 1),
-	pages : FW.Xpath('//div[@class="ref_on_top"]')
-					.text().trim().match(/\d+(?:–\d+)?$/),
-	date : FW.Xpath('//div[@id="content"]//td[contains(text(),"Published")]')
-					.text().trim().match(/Published\s+(.+)/,1),
-	ISSN : '1565-8872',
-	attachments : [
+	let refOnTop = (text(doc, 'div.ref_on_top') || '').trim();
+	item.volume = match(refOnTop, /volume (\d+)/i);
+	item.pages = match(refOnTop, /(\d+(?:–\d+)?)$/);
+
+	let date = Array.from(doc.querySelectorAll('div#content td')).find(td => td.textContent.includes('Published'));
+	if (date) item.date = match(date.textContent, /Published\s+(.+)/);
+	item.ISSN = '1565-8872';
+
+	item.attachments = [
 		{
-			url : FW.Url().replace(/(abstracts|html)/i,'LetterSize')
-						.replace(/\.html.*$/,'.pdf'),
-			title : 'Full text PDF',
-			type : 'application/pdf'
+			url: url.replace(/(abstracts|html)/i, 'LetterSize').replace(/\.html.*$/, '.pdf'),
```

Comment:
I guess this does not work in the French version. Can we make this more general? (Personally I would also prefer the A4 format over the letter.)